cmake/prescan.cmake

```diff
@@ -12,22 +12,43 @@
 # full build or the information will be wrong.
 ####
 
-# NOTE: **ensure** that list properties are added by hand in the actual call. Otherwise they are expanded and break
-# the call below.
-set(NEEDED_PROPERTIES
-    FPRIME_CONFIG_DIR
-    FPRIME_AC_CONSTANTS_FILE
-    FPRIME_ENVIRONMENT_FILE
-    FPRIME_SETTINGS_FILE
-    FPRIME_PROJECT_ROOT
-    FPRIME_FRAMEWORK_PATH
-    CMAKE_TOOLCHAIN_FILE
-    CMAKE_BUILD_TYPE
-    CMAKE_DEBUG_OUTPUT
-    FPRIME_USE_STUBBED_DRIVERS
-    FPRIME_USE_BAREMETAL_SCHEDULER
-    FPP_TOOLS_PATH
-    BUILD_TESTING
+set(EXCLUDED_CACHE_VARIABLES
```

Comment:
Out of curiosity, how did you select which variables had to be excluded from the cache?

Reply:
Ideally I wanted to pass everything.  However, CMake emitted warnings about unneeded/unused variables. I added those to the list along with the ones we explicitly set in the call below.